app/src/main/java/co/nimblehq/alpine/sample/MainActivity.kt

```diff
@@ -1,12 +1,80 @@
 package co.nimblehq.alpine.sample
 
-import androidx.appcompat.app.AppCompatActivity
+import android.Manifest.permission.CAMERA
+import android.content.Intent
+import android.content.pm.PackageManager.PERMISSION_GRANTED
+import android.net.Uri
 import android.os.Bundle
+import android.provider.Settings
+import android.view.View
+import android.widget.Button
+import android.widget.Toast
+import androidx.activity.ComponentActivity
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.appcompat.app.AlertDialog
+import androidx.core.content.ContextCompat.checkSelfPermission
 import co.nimblehq.alpine.R
 
-class MainActivity : AppCompatActivity() {
+class MainActivity : ComponentActivity(), View.OnClickListener {
+
+    private val requestPermissionLauncher =
+        registerForActivityResult(
+            ActivityResultContracts.RequestPermission()
+        ) { isGranted: Boolean ->
+            if (isGranted) {
+                navigateToCamera()
+            } else {
+                showRationale()
+            }
+        }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
```

Comment:
Any reason we didn't go with View Binding here? 🤔

Reply:
@ryan-conway for brevity. IMO, it'd be better to keep the sample as basic as possible. What do you think?

Reply:
@AVI5HEK makes sense since its a sample project and not the actual library itself 👍🏻